tree/ntuple/v7/inc/ROOT/RColumn.hxx

```diff
@@ -48,23 +48,52 @@ private:
     * the offset column with index 0 and the character value column with index 1.
     */
    std::uint32_t fIndex;
-   RPageSink *fPageSink;
-   RPageSource *fPageSource;
+   RPageSink *fPageSink = nullptr;
+   RPageSource *fPageSource = nullptr;
    RPageStorage::ColumnHandle_t fHandleSink;
    RPageStorage::ColumnHandle_t fHandleSource;
-   /// Open page into which new elements are being written
-   RPage fHeadPage;
+   /// A set of open pages into which new elements are being written. The pages are used
+   /// in rotation. They are 50% bigger than the target size given by the write options.
+   /// The current page is filled until the target size, but it is only committed once the other
+   /// head page is filled at least 50%. If a flush occurs earlier, a slightly oversize, single
+   /// page will be committed.
+   RPage fHeadPage[2];
+   /// Index of the current head page
+   int fHeadPageIdx = 0;
+   /// For writing, the targeted page size as given by the write options.
+   /// We ensure this value to be >= 2 in Connect() so that we have meaningful
+   /// "page full" and "page half full" events when writing the page.
+   std::uint32_t fApproxNElementsPerPage = 0;
    /// The number of elements written resp. available in the column
-   NTupleSize_t fNElements;
+   NTupleSize_t fNElements = 0;
    /// The currently mapped page for reading
    RPage fCurrentPage;
    /// The column id is used to find matching pages with content when reading
-   ColumnId_t fColumnIdSource;
+   ColumnId_t fColumnIdSource = kInvalidColumnId;
    /// Used to pack and unpack pages on writing/reading
    std::unique_ptr<RColumnElementBase> fElement;
 
    RColumn(const RColumnModel &model, std::uint32_t index);
 
+   /// Used in Append() and AppendV() to switch pages when the main page reached the target size
+   /// The other page has been flushed when the main page reached 50%.
+   void SwapHeadPages() {
+      fHeadPageIdx = 1 - fHeadPageIdx;
```

Comment:
This seems to indicates that `fHeadPageIdx` is alternating between positive and negative values.
Is that the intent?
If so, then the part of the code that insure that `Append` and `AppendV` are called only when `fHeadPageIdx` is positive is not obvious (i.e. did not find it yet).
What am I missing?

Reply:
humm ... I missed that `0 <= fHeadPageIdx <= 1` ... and thus `1 - fHeadPageIdx` is 'return other value` ... maybe it is worth either adding a comment (at use use of that pattern) or hide it being a function (`ReturnOtherValue(...)`)

Reply:
[Comment added](https://github.com/root-project/root/pull/8703/commits/421a04fe523d1a424dd88618aff76f6cec21bada)